applications/GeoMechanicsApplication/python_scripts/geomechanics_T_solver.py

```diff
@@ -0,0 +1,170 @@
+
+        self.main_model_part.ProcessInfo.SetValue(KratosGeo.DT_TEMPERATURE_COEFFICIENT, 1.0)
+
+        if scheme_type.lower() == "newmark" or scheme_type.lower() == "newmark_flow":
+            theta = self.settings["newmark_theta"].GetDouble()
+            KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver, solution_type", solution_type)
+            if solution_type.lower() == "transient-heat-transfer" or solution_type.lower() == "transient_heat_transfer":
+                KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver, scheme", "Newmark Transient heat transfer.")
+                scheme = KratosGeo.NewmarkTScheme(theta)
+            elif solution_type.lower() == "steady-state-heat-transfer" or solution_type.lower() == "steady_state_heat_transfer":
+                raise Exception("Steady state heat transfer calculations are not supported.")
+            else:
+                raise Exception("Undefined solution type", solution_type)
+
+        elif scheme_type.lower() == "backward_euler":
+            if solution_type.lower() == "transient-heat-transfer" or solution_type.lower() == "transient_heat_transfer":
+                KratosMultiphysics.Logger.PrintInfo("GeoMechanics_T_Solver, scheme", "Backward Euler Transient heat transfer.")
+                scheme = KratosGeo.BackwardEulerTScheme()
+            elif solution_type.lower() == "steady-state-heat-transfer" or solution_type.lower() == "steady_state_heat_transfer":
+                raise Exception("Steady state heat transfer calculations are not supported.")
+            else:
+                raise Exception("Undefined solution type", solution_type)
+
+        else:
+            raise Exception("Apart from Newmark and Backward Euler, no other scheme_type is available for thermal calculations.")
+
+        return scheme
+
+
+
+    def _ConstructConvergenceCriterion(self, convergence_criterion):
+        if convergence_criterion.lower() == "temperature_criterion":
+            return self._GetTemperatureCriterion()
+        elif convergence_criterion.lower() == "residual_criterion":
+            return self._GetResidualCriterion()
+        elif convergence_criterion.lower() == "and_criterion":
+            temperature = self._GetTemperatureCriterion()
+            residual = self._GetResidualCriterion()
+            return KratosMultiphysics.AndCriteria(residual, temperature)
+        elif convergence_criterion.lower() == "or_criterion":
+            temperature = self._GetTemperatureCriterion()
+            residual = self._GetResidualCriterion()
+            return KratosMultiphysics.OrCriteria(residual, temperature)
+        else:
+            err_msg = "The requested convergence criterion \"" + convergence_criterion + "\" is not available!\n"
+            err_msg += "Available options are: \"temperature_criterion\", \"residual_criterion\", \"and_criterion\", \"or_criterion\""
+            raise Exception(err_msg)
+
+    def _GetTemperatureCriterion(self):
```

Comment:
Just a remark: I understand that for consistency this function is called `_Get...`, although it _creates_ something new. Probably a better name would have been `_MakeTemperatureCriterion`, but then it deviates from the rest :-s